platform.js

```diff
@@ -154,6 +154,19 @@ class App {
 
     const dest = path.join(this.originalResourcesDir, 'app.asar')
     debug(`Running asar with the options ${JSON.stringify(asarOptions)}`)
+
+    if (asarOptions.filename) {
+      const src = path.resolve(asarOptions.filename)
+
+      return fs.stat(src)
+        .then(stat => {
+          const target = stat.isFile() ? dest : this.originalResourcesDir
+          debug(`Copying asar: ${src} to ${target}`)
+          return fs.copy(src, target, {overwrite: false, errorOnExist: true})
+        })
+        .then(() => fs.remove(this.originalResourcesAppDir))
```

Comment:
I think for efficiency purposes, if `asar.filename` is set, we shouldn't even copy the app directory in the first place. Then we can get rid of this `fs.remove` call.

Reply:
I started to go down that route, but tried to keep the change minimal. I'll short circuit that.